modules/core/src/commonMain/kotlin/fr/acinq/lightning/channel/states/Channel.kt

```diff
@@ -78,12 +78,15 @@ sealed class ChannelState {
             is Syncing -> this@ChannelState.state
             else -> this@ChannelState
         }
-        maybeSignalSensitiveTask(oldState, newState)
-        return when {
-            // we only want to fire the PaymentSent event when we transition to Closing for the first time
-            oldState is ChannelStateWithCommitments && oldState !is Closing && newState is Closing -> emitClosingEvents(oldState, newState)
-            else -> emptyList()
+        @Suppress("NAME_SHADOWING")
+        val newState = when (newState) {
```

Comment:
The shadowing is admittedly ugly, but I find it safer than using a different variable, as it forces us to use the correct one.